### ISSUE TITLE
 v0.2.1 could break existing consumers as Js.Result and Belt.Result are incompatible

### ISSUE THREAD

**Issue by jchavarri** [glennsl/rebase#2]
`Js.Result` and `Belt.Result` don't resolve to the same type. The upgrade to v0.2.1 was tagged as patch, but it can actually break existing consumers as the API is changing to different types.

**Comment by glennsl** [comment]
You're right, it should have been 0.3.0 at least. Unless you think it's a big problem I'd prefer to avoid a messy revert and re-release though. Otherwise not sure exactly what you have in mind.

I do wish these things would be given some more patience, thought and consideration upstream, and perhaps a warning-less soft deprecation could be performed before the hard deprecation to reduce the churn and noise by giving library authors the opportunity to fix it before the alarm goes off. But this might just be what you get when the developer does not actually use what they're making, but are focused on other interests instead.
